data/modules/TradeShips/Trader.lua

```diff
@@ -0,0 +1,355 @@
+-- Copyright © 2008-2021 Pioneer Developers. See AUTHORS.txt for details
+-- Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
+
+local e = require 'Equipment'
+local Engine = require 'Engine'
+local Game = require 'Game'
+local Ship = require 'Ship'
+local ShipDef = require 'ShipDef'
+local Space = require 'Space'
+local Timer = require 'Timer'
+
+local Core = require 'modules.TradeShips.Core'
+
+local Trader = {}
+-- this module contains functions that work for single traders
+
+Trader.addEquip = function (ship)
+	local ship_type = ShipDef[ship.shipId]
+
+	-- add standard equipment
+	ship:AddEquip(e.hyperspace['hyperdrive_'..tostring(ship_type.hyperdriveClass)])
+	if ShipDef[ship.shipId].equipSlotCapacity.atmo_shield > 0 then
+		ship:AddEquip(e.misc.atmospheric_shielding)
+	end
+	ship:AddEquip(e.misc.radar)
+	ship:AddEquip(e.misc.autopilot)
+	ship:AddEquip(e.misc.cargo_life_support)
+
+	-- add defensive equipment based on lawlessness, luck and size
+	local lawlessness = Game.system.lawlessness
+	local size_factor = ship.freeCapacity ^ 2 / 2000000
+
+	if Engine.rand:Number(1) - 0.1 < lawlessness then
+		local num = math.floor(math.sqrt(ship.freeCapacity / 50)) -
+		ship:CountEquip(e.misc.shield_generator)
+		if num > 0 then ship:AddEquip(e.misc.shield_generator, num) end
+		if ship_type.equipSlotCapacity.energy_booster > 0 and
+			Engine.rand:Number(1) + 0.5 - size_factor < lawlessness then
+			ship:AddEquip(e.misc.shield_energy_booster)
+		end
+	end
+
+	-- we can't use these yet
+	if ship_type.equipSlotCapacity.ecm > 0 then
+		if Engine.rand:Number(1) + 0.2 < lawlessness then
+			ship:AddEquip(e.misc.ecm_advanced)
+		elseif Engine.rand:Number(1) < lawlessness then
+			ship:AddEquip(e.misc.ecm_basic)
+		end
+	end
+
+	-- this should be rare
+	if ship_type.equipSlotCapacity.hull_autorepair > 0 and
+		Engine.rand:Number(1) + 0.75 - size_factor < lawlessness then
+		ship:AddEquip(e.misc.hull_autorepair)
+	end
+end
+
+Trader.addCargo = function (ship, direction)
+	local imports = Core.params.imports
+	local exports = Core.params.exports
+	local total = 0
+	local empty_space = math.min(ship.freeCapacity, ship:GetEquipFree("cargo"))
+	local size_factor = empty_space / 20
+	local ship_cargo = {}
+
+	if direction == 'import' and #imports == 1 then
+		total = ship:AddEquip(imports[1], empty_space)
+		ship_cargo[imports[1]] = total
+	elseif direction == 'export' and #exports == 1 then
+		total = ship:AddEquip(exports[1], empty_space)
+		ship_cargo[exports[1]] = total
+	elseif (direction == 'import' and #imports > 1) or
+		(direction == 'export' and #exports > 1) then
+
+		-- happens if there was very little space left to begin with (eg small
```

Comment:
Instead of having all of these branches for import vs export, prefer to assign the list of cargo to a single variable, e.g.:
```lua
local cargoTypes = direction == 'import' and Core.params.imports or Core.params.exports

if #cargoTypes == 1 then
    total = ship:AddEquip(cargoTypes[1], empty_space)
    -- ...
elseif #cargoTypes > 1 then
    local cargo_type = cargoTypes[Engine.rand:Integer(1, #cargoTypes)]
    -- ...
end
```

Reply:
Done!